tests/middleware/test_exception_handler_middleware.py

```diff
@@ -118,3 +120,36 @@ async def after_exception_hook_handler(exc: Exception, scope: "Scope", state: "S
         response = client.get("/test")
         assert response.status_code == HTTP_500_INTERNAL_SERVER_ERROR
         assert client.app.state.called
+
+
+def test_exception_handler_middleware_debug_logging(caplog: "LogCaptureFixture") -> None:
+    @get("/test")
+    def handler() -> None:
+        raise ValueError("Test debug exception")
+
+    # Debug On -> Exception is logged
+    with caplog.at_level("DEBUG"), create_test_client(handler, logging_config=LoggingConfig()) as client:
+        client.app.debug = True
+        response = client.get("/test")
+        assert response.status_code == HTTP_500_INTERNAL_SERVER_ERROR
+        assert "Internal Server Error" in caplog.text
+        assert "ValueError" in response.text
+        assert "Test debug exception" in response.text
+
+    # Debug Off -> Exception is not logged
+    with caplog.at_level("INFO"), create_test_client(handler, logging_config=LoggingConfig()) as client:
+        client.app.debug = False
+        response = client.get("/test")
+        assert response.status_code == HTTP_500_INTERNAL_SERVER_ERROR
+        assert "Exception in ASGI application" not in caplog.text
```

Comment:
since this was changed, this condition will alway be true. so you should update this line.